File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/domain/error/DomainError.java
package io.github.malczuuu.lemur.app.domain.error;

public enum DomainError implements ErrorType {
  INTERNAL_ERROR(ErrorCategory.INTERNAL_ERROR),
  CONCURRENT_MODIFICATION(ErrorCategory.CONFLICT),
  ;

  private final ErrorCategory category;

  DomainError(ErrorCategory category) {
    this.category = category;
  }

  @Override
  public ErrorCategory getCategory() {
    return category;
  }

  @Override
  public String getType() {
    return name();
  }
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/domain/error/DomainError.kt
package io.github.malczuuu.lemur.app.domain.error

enum class DomainError(override val category: ErrorCategory) : ErrorType {
    INTERNAL_ERROR(ErrorCategory.INTERNAL_ERROR),
    CONCURRENT_MODIFICATION(ErrorCategory.CONFLICT),
    ;

    override val type: String
        get() = name
}
